### PR TITLE
Retain sort order across paged case lists

### PR DESCRIPTION
**Pull request by tomsmyers**
https://trello.com/c/blHohWHb/83-sort-order-of-paginated-case-lists-does-not-get-applied-across-pages

## What does this pull request do?

uses the API to sort referrals in the PP dashboards. stores the user defined sort in redis. disables client-side table sorting per-page.

a bunch of integration tests are removed - this is because we can no longer test the sorting, it's done in the backend. however a new test has been added that verifies the presenter class is rendering the headings correctly based on requested sort order.

## What is the intent behind these changes?

retain sort order across pages

### CODE REVIEW

File: server/routes/probationPractitionerReferrals/dashboardPresenter.ts [hunk 1]
```diff
@@ -17,36 +17,63 @@ export default class DashboardPresenter {
...
+        sort: sortField === 'intervention.dynamicFrameworkContract.primeProvider.name' ? ariaSortOrder : 'none',
+        persistentId: 'intervention.dynamicFrameworkContract.primeProvider.name',
+      },
+      this.showAssignedCaseworkerColumn
+        ? null
+        : {
+            text: 'Caseworker',
+            sort: 'none',
+            persistentId: null,
+            classes: { color: 'blue' },
```
Comment (tomsmyers) [inline server/routes/probationPractitionerReferrals/dashboardPresenter.ts:72]: remove

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 2]
```diff
@@ -48,36 +48,65 @@ export default class ProbationPractitionerReferralsController {
...
+
+    // if a query param is passed, always use it (and store it, later). otherwise, try
+    // to get the sort order from the user data service, falling back to the default.
+    const sortQueryParam = req.query.sort
+    const primarySort =
+      sortQueryParam !== undefined
+        ? (sortQueryParam as string)
+        : (await this.userDataService.retrieve(userId, userSortKey)) ?? defaultPrimarySort
+
+    const sort = primarySort.startsWith('sentAt') ? [primarySort] : [primarySort, 'sentAt,ASC']
```
Comment (tomsmyers) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:103]: can we get the API to just apply this a default secondary sort?

File: assets/sass/components/_sortable-table.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.non-sortable__sortable-table-header {
+  color: map-get($_govuk-colour-palette-legacy, 'blue'),;
```
Comment (tomsmyers) [inline assets/sass/components/_sortable-table.scss:2]: in the end i went for the legacy color to match

File: browser/mojServerSideSortableTable.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        // we need to figure out which of these was actually clicked. We do that
+        // by finding out which of the columns is sorted (by looking at the
+        // headers’ aria-sort attributes).
+
+        const header = mutation.target as HTMLElement
+
+        const newAriaSort = this.createARIASort(header.getAttribute('aria-sort'))
+
+        if (newAriaSort === null) {
+          console.warn('Unrecognised aria-sort attribute')
```
Comment (sldblog) [inline browser/mojServerSideSortableTable.ts:64]: _nit_ This warning could live in `createARIASort` to keep them closer together

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 5]
```diff
@@ -39,42 +40,69 @@ export default class ProbationPractitionerReferralsController {
...
+      'Completed cases',
+      'ppCompletedCases',
+      pageSize
+    )
   }
 
   async showCancelledCases(req: Request, res: Response): Promise<void> {
     const pageSize = config.apis.interventionsService.dashboardPageSize.pp.cancelledCases
-    await this.showDashboard(req, res, { cancelled: true }, 'Cancelled cases', pageSize)
+    await this.showDashboard(req, res, { cancelled: true }, 'Cancelled cases', 'ppCancelledCases', pageSize)
```
Comment (sldblog) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:80]: 💭 we have `type PPDashboardType = 'Open cases' | 'Unassigned cases' | 'Completed cases' | 'Cancelled cases'` -- I wonder if this new ID could be woven into that type, given they define the same concept?

**Review by sldblog — APPROVED** [review]
LGTM, no blockers

**Pull request merged**